Program.cs

```diff
@@ -1,7 +1,33 @@
-﻿namespace Training {
-    internal class Program {
-        static void Main (string[] args) {
-            Console.WriteLine ("Hello, World!");
-        }
-    }
-}
\ No newline at end of file
+﻿internal class Program {
+   private static void Main (string[] args) {
+      int a = 1;
+      int z = 101;
+      Random r = new ();
+      int n = r.Next (a, z);
+
+      Console.WriteLine ($"GUESS RANDOM NUMBER FROM {a} TO {z}:");
+      //Console.WriteLine (n);
+      int attempt = 0;
+      int p = 0;
+      while (p != n) {
+         // get user input as string
+         Console.Write ("Guess number: ");    
+         // Type cast - convert string to int
+         if (int.TryParse (Console.ReadLine (), out p)) {
+            // check if non-positive and print msg
+            if (p < a || p > z)
+               Console.WriteLine ("Enter the number within the range.");

```

Comment:
you should "continue" to skip the remaining statements as those will also be executed if the number is not in the range.